chanzuckerberg/single-cell-data-portal
# fix: Avoid concurrency name collisions when deploying rdevs.

## Reason for Change

jobs are being canceled because the concurrency name use is not correctly populated. This results in jobs being unintentionally canceled 
<img width="967" alt="Screenshot 2024-03-20 at 1 40 22 PM" src="https://github.com/chanzuckerberg/single-cell-data-portal/assets/1429913/d2cac79e-68b2-4f53-b34b-346d5fbbfdf7">

## Changes

- use a work around to use STACK_NAME for concurrency group. concurrency cannot use the `env` context but it can use the `needs` context.

## Testing steps

- Run build image in two different PR simultaneously, and verify they both succeed.
